lib/Sema/CSSimplify.cpp

```diff
@@ -960,9 +960,34 @@ ConstraintSystem::TypeMatchResult constraints::matchCallArguments(
 
   for (unsigned paramIdx = 0, numParams = parameterBindings.size();
        paramIdx != numParams; ++paramIdx){
-    // Skip unfulfilled parameters. There's nothing to do for them.
-    if (parameterBindings[paramIdx].empty())
+    // If a parameter is unfulfilled, validate the default argument if it is a
+    // magic literal expression.
+    if (parameterBindings[paramIdx].empty()) {
+      if (!isa<FuncDecl>(callee))
+        continue;
+
+      auto allParams = dyn_cast<FuncDecl>(callee)->getParameters()->getArray();
+      auto param = allParams[paramIdx];
+
+      assert(param->isDefaultArgument());
+
+      auto defaultValueExpr = param->getDefaultValue();
+
+      if (!isa<MagicIdentifierLiteralExpr>(defaultValueExpr))
+        continue;
+
+      if (!cs.hasType(defaultValueExpr)) {
```

Comment:
That's actually incorrect - the type of default value is not going to be cached in the constraint system because it's not a part of AST when constraints are generated, you have to use `defaultValueExpr->getType()` here.